gammapy/scripts/cta_irf.py

```diff
@@ -0,0 +1,105 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import numpy as np
+import astropy.units as u
+from astropy.table import Table
+from ..utils.fits import get_hdu
+from ..irf import EffectiveAreaTable2D
+from ..background import Cube
+from ..irf import EnergyDispersion2D
+from ..irf import EnergyDependentMultiGaussPSF
+
+__all__ = [
+    'CTAIrf',
+]
+
+
+class CTAIrf(object):
+    """Contain CTA instrument response functions.
+
+    Class handling CTA instrument response function. For now
+    we use the production 2 of the CTA IRF
+    (https://portal.cta-observatory.org/Pages/CTA-Performance.aspx)
+    adapted from the ctools
+    (http://cta.irap.omp.eu/ctools/user_manual/getting_started/response.html).
+    The IRF format should be complient compliant with the one discussed
+    at http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/.
+    Waiting for a new public production of the CTA IRF,
+    we'll fix the missing pieces.
+
+    Parameters
+    ----------
+    area_2d : `~gammapy.irf.EffectiveAreaTable2D`
+        2D Effective Area Table
+    bcg_cube : `~gammapy.background.Cube`
+        Cube container class with scheme fixed to `bg_cube`
+    edisp_2d : `~gammapy.irf.EnergyDispersion2D`
+        Offset-dependent energy dispersion matrix
+    psf_mgauss : `~gammapy.irf.EnergyDispersion2D`
+        Position-dependent multi-Gauss PSF
+    """
+
+    def __init__(self, area_2d=None, bkg_cube=None, e_disp=None, psf_mgauss=None):
+        self.area_2d = area_2d
+        self.bkg_cube = bkg_cube
+        self.e_disp_2d = e_disp_2d
+        self.psf_mgauss = psf_mgauss
+
+    @classmethod
+    def from_fits(cls, irf_path):
+        """
+        Auxiliary constructor from a fits file
+
+        Parameters
+        ----------
+        path : `str`
+            Path of the file containing the IRF
+        """
+        # Dealing with effective area and fix headers
+        table_hdu_area = get_hdu(irf_path + '[EFFECTIVE AREA]')
+        table_area = Table(table_hdu_area.data)
+
+        table_area['ENERG_LO'].unit = u.TeV
+        table_area['ENERG_HI'].unit = u.TeV
+        table_area['THETA_LO'].unit = u.deg
+        table_area['THETA_HI'].unit = u.deg
+        table_area['EFFAREA'].unit = u.m * u.m
+        table_area['EFFAREA_RECO'].unit = u.m * u.m
+
+        area_2d = EffectiveAreaTable2D.from_table(table_area)
+
+        # Dealing with background and fix headers
+        table_hdu_bkg = get_hdu(irf_path + '[BACKGROUND]')
+
+        table_hdu_bkg.header['TTYPE7'] = 'Bgd'
+        table_hdu_bkg.header['TUNIT7'] = '1 / (MeV s sr)'
+
+        bkg_cube = Cube.from_fits_table(table_hdu_bkg, scheme='bg_cube')
```

Comment:
This is failing with a `TypeError` on some travis-ci builds:
https://travis-ci.org/gammapy/gammapy/jobs/147251046#L940

I'm not getting this error locally, not sure yet what's causing it ...

!?